internal/pkg/overlay/overlay.go

```diff
@@ -496,12 +496,7 @@ func RenderTemplateFile(fileName string, data TemplateStruct) (
 		"softlink":     softlink,
 		"readlink":     filepath.EvalSymlinks,
 		"IgnitionJson": func() string {
-			str := createIgnitionJson(data.ThisNode)
-			if str != "" {
-				return str
-			}
-			writeFile = false
-			return ""
+			return createIgnitionJson(data.ThisNode)
```

Comment:
Looks much simpler this way. I think I added the checks as security measurement, but forgot why